src/runtime/metal/metal_common.h

```diff
@@ -42,9 +42,37 @@
 
 #include "../workspace_pool.h"
 
+#define AUTORELEASEPOOL tvm::runtime::metal::AutoReleasePoolWrapper::GetInstance() << [&]()
+
 namespace tvm {
 namespace runtime {
 namespace metal {
+class AutoReleasePoolWrapper {
+ public:
+  static AutoReleasePoolWrapper& GetInstance() {
```

Comment:
Not sure if this construction safe been used from several libraries. Even if we do not have several modules, it's better to move body of the function into cpp file

Reply:
Done